catalog/task/push-snapshot/0.4/push-snapshot.yaml

```diff
@@ -0,0 +1,71 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: push-snapshot
+  labels:
+    app.kubernetes.io/version: "0.4"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Tekton task to push snapshot images to an image registry using `skopeo copy`
+  params:
+    - name: mappedSnapshot
+      description: JSON string representing the Snapshot
+      type: string
+    - name: tag
+      description: Default tag to use if mapping entry does not contain a tag
+      type: string
+      default: "latest"
+    - name: retries
+      description: Retry copy N times.
+      type: string
+      default: "0"
+  steps:
+    - name: skopeo-copy
+      image:
+        quay.io/hacbs-release/release-base-image@sha256:9e7fd1a3ccf0d2c8077f565c78e50862a7cc4792d548b5c01c8b09077e6d23a7
+      script: |
+        #!/usr/bin/env bash
+        set -eux
+        application=$(jq -r '.application' <<<'$(params.mappedSnapshot)')
+        printf 'Beginning "%s" for "%s"\n\n' "$(context.task.name)" "$application"
+        while read line;
+        do
+          # Create array with component values
+          typeset -A data
+          while IFS== read -r key value; do
+            data["$key"]="$value"
+          done < <(jq -r '. | to_entries | .[] | .key + "=" + .value' <<<"$line")
+          # take tag if present in mapping file, otherwise take ReleaseStrategy default
+          data[tag]="${data[tag]:=$(params.tag)}"
+          # default to latest if no tag specified in mapping file or ReleaseStrategy
+          data[tag]="${data[tag]:=latest}"
```

Comment:
the param default is latest. So basically what we are saying here is 
if params.tag was changed from the default (latest) to be the empty string *and* there is none in the mapping file, then use latest
Is that correct? It makes sense logically, but it is kind of unfortunate we have to guard against someone changing the parameter from the default to the empty string and then us putting the default back in

Reply:
Yeah, this seems redundant. I think we should be fine with the fact that if someone sets the param to "", it will break. It's an invalid input, just like e.g. supplying a broken snapshot json or anything else.

Reply:
ok so the consensus is the remove `line 45`?

Reply:
Yes, I would remove it. But I'm not entirely sure if @johnbieren meant the same, let's see :)

Reply:
I am in favor of removing it

Reply:
removed